app/controllers/vendor_api/interviews_controller.rb

```diff
@@ -3,12 +3,17 @@ class InterviewsController < VendorAPIController
     include ApplicationDataConcerns
     include APIValidationsAndErrorHandling
 
+    rescue_from InterviewWorkflowConstraints::WorkflowError, with: :render_workflow_error
+    rescue_from NotModifiedError, with: :render_not_modified
+    rescue_from InvalidProviderCode, with: :render_invalid_provider_code
+    rescue_from InvalidDateError, with: :render_invalid_date
```

Comment:
We should be rendering these as validation errors. We may want to consider using a new error type for the `NotModifiedError`, but even then I think we should address separately as it's not part of the current draft.

```suggestion
    rescue_from InterviewWorkflowConstraints::WorkflowError, with: :render_validation_error
    rescue_from NotModifiedError, with: :render_validation_error
    rescue_from InvalidProviderCode, with: :render_validation_error
    rescue_from InvalidDateError, with: :render_validation_error
```

Reply:
I'll add something like

``` ruby
    def render_error_as_json(e)
      render status: :unprocessable_entity, json: {
        errors: [
          {
            error: e.class.to_s,
            message: e.message,
          },
        ],
      }
    end
```

as `render_validation_error` just prints a string, not json.